core/retries/src/main/java/software/amazon/awssdk/retries/internal/DefaultAwareRetryStrategy.java

```diff
@@ -22,25 +22,21 @@
  * Identify a {@link RetryStrategy} that has the capacity to work with sets of default retry predicates.
  */
 @SdkProtectedApi
-public interface DefaultAwareRetryStrategy {
+public interface DefaultAwareRetryStrategy extends RetryStrategy {
 
     /**
-     * Check if the specified set of predicates, identified by their name, has been added to this retry strategy
-     * @param defaultPredicatesName the name of the set of predicates to check
-     * @return true if the named set of predicate
+     * Add the specified defaults to this retry strategy
+     * @param retryStrategyDefaults the defaults to add to this strategy
+     * @return a new retry strategy containing the specified defaults.
      */
-    default boolean shouldAddDefaults(String defaultPredicatesName) {
-        return false;
-    }
+    DefaultAwareRetryStrategy addDefaults(RetryStrategyDefaults retryStrategyDefaults);
 
     interface Builder {
 
         /**
-         * Identify the Builder has having the specified set of predicate added to it.
-         * @param defaultPredicatesName the name of the set of predicate
+         * Identify the Builder as having the specified defaults to it.
+         * @param defaultPredicatesName the name the defaults to mark as added
          */
-        default void markDefaultAdded(String defaultPredicatesName) {
-
-        }
+        void markDefaultAdded(String defaultPredicatesName);
```

Comment:
Is this method needed anymore?

Reply:
we do if we want to avoid having the core retry classes reference `BaseRetryStrategy` like you proposed eariler, which I think is the better approach. For example, `AwsRetryStrategy` needs to be able to mark defaults as being added when instantiated directly, to not duplicate them when the default client builder tries to add them later at client build time.

Reply:
Got it, I imagined `AwsRetryStrategy` using `addDefaults` on instantiation to ensure they're marked as already included. I can see why that's weird now. For some reason I also imagined `addDefaults` being on the builder, not on the retry strategy itself.